### ISSUE TITLE
Suffix of words (plus next 5 chars) mistakenly being converted to emojis

### ISSUE THREAD

**Issue by e-maya** [vector-im/element-web#22401]
This is an issue reported by @Tezcatlipoca:matrix.org:

THIS AFFECTS BOTH WEB AND DESKTOP VERSIONS

**With "Automatically replace plain text Emoji" activated, try to write any word with a "xo" suffix, followed by at least 6 more characters. An example in Portuguese:**

"embaixo da casa"

It should just stay as it is while I'm typing (i.e. **"embaixo da casa"**), but instead Element  converts it automatically to **"embai😵asa"**, not only converting the mapped string (in the example, "xo") - which it shouldn't be doing anyway -, but also "eating" the next 5 characters in the process.

This bug can be reproduced with any other mapped plaintext->emoji strings, like "xD" or "XD" for exemple. So if you write "WordXD whatever", the 6th character will trigger the bug. "xo" seems to be the more critical AFAIK, for there are words that are really common in my native language (Portuguese) which are directly affected by this bug.

In short, the suffixes of words should never be interpreted as candidates for conversion to emojis.

**Comment by kerryarchibald** [comment]
Duplicate https://github.com/vector-im/element-web/issues/19472
